data/data-pipeline/data_pipeline/etl/score/etl_utils.py

```diff
@@ -108,3 +110,174 @@ def floor_series(series: pd.Series, number_of_decimals: int) -> pd.Series:
     )
 
     return floored_series
+
+
+def _create_df_from_yaml_contents(
+    fields_list_from_yaml: list,
+    fields_to_store_in_codebook: list,
+) -> pd.DataFrame:
+    """Helper function to create a dataframe from yaml fields
+
+    This function does:
```

Comment:
I'm sorry, but I'm having a hard time understanding your steps 1-3 in this documentation. Do you mind rewriting this to try to be more clear? There's a lot that seems a little handwavey in your explanation.

Reply:
Updated